### ISSUE TITLE
On Android Share.open result me only success. I am expecting the success + app details

### ISSUE THREAD

**Issue by PradeepVL** [react-native-share/react-native-share#1615]
### Steps to reproduce
  await Share.open(options)
        .then((res) => {
          console.log(res);
        })
        .catch((err) => {
          err && console.log(err);
        });
Log is : {success: true}

### Expected behavior
{success: true , app:'com.instagram.android'}

### Actual behavior
{success: true }
app key is missing

### Environment
- **React Native version**:0.71.14
- **React Native platform + platform version**: Pixel 2 Android 15 

### react-native-share
Latest

### Link to repo (highly encouraged)
https://github.com/

**Comment by Salman9000** [comment]
Hello, i am getting {"message": "com.instagram.android", "success": true}, but the instagram app flashes and then closes. What to do?

**Comment by PradeepVL** [comment]
> Hello, i am getting {"message": "com.instagram.android", "success": true}, but the instagram app flashes and then closes. What to do?

@Salman9000 What are you using Share.shareSingle or Share.open?. Could you please add your snippet

**Comment by PradeepVL** [comment]
> social: Share.Social.INSTAGRAM_STORIES,

OK, I guess because you are adding key social with value on options with shareSingle, you are able to receive message or app valu

**Comment by x1frm** [comment]
For me shareSingle always resolves immediately with `{"message": "<selectedApp>", "success": true}`. There is no way to understand if the user declined the dialog or not, or if any other error happened.

Expo 51.0.35 managed workflow, RN 0.74.5

```
/* example options:
               {
                    url: `data:video/mp4;base64,${base64String}`,
                    social: Social.Instagram,
                    title: 'Video',
                    message: 'Video'
                }
*/
export const share = async (options: Partial<ShareSingleOptions>) => {
    try {
        if (options.social) {
            const shareOptions = {
                type: 'video/mp4',
            };

            await Share.shareSingle({ ...shareOptions, ...options } as ShareSingleOptions);
        } else {
            await Share.open(options);
        }

        if (shouldShowToast) {
            showToast(STRINGS.VIDEO_PLAYER.toast.share);
        }
    } catch (err) {
        // We don't want to log an error if the user cancels the share
        if (err instanceof Error && err.message === 'User did not share') {
            return;
        }

        logger.error(err);
    }
};
```

**Comment by PradeepVL** [comment]
> Share.open

What about Share.open?

**Comment by PradeepVL** [comment]
@mikehardy Could you please look into this.

**Comment by mikehardy** [comment]
@PradeepVL no, I can't. All out of free time to donate to unknown people at the moment. You'll be best served adding some logging statements to the relevant native code and examining the platform API details to see if what you want is possible. I suspect the information isn't returned, and what you want is not possible. But you'll need to prove that to yourself with the API investigation + logging. If it is possible you'll be in a good position to post a PR then making what you want possible happen. Cheers
